platform/bukkit/plugin/src/main/java/me/lokka30/treasury/plugin/bukkit/BukkitTreasuryPlugin.java

```diff
@@ -129,24 +129,46 @@ private String colorizeHex(@NotNull String message) {
 
     @Override
     public void info(String message) {
-        plugin.getServer().getConsoleSender().sendMessage("[Treasury] " + colorize(message));
+        if (BukkitVendor.isPaper()) {
+            plugin.getServer().getConsoleSender().sendMessage("[Treasury] " + colorize(message));
+        } else {
+            plugin.getLogger().info(colorize(message));
+        }
     }
 
     @Override
     public void warn(String message) {
-        plugin.getServer().getConsoleSender().sendMessage(
-                "&e[WARNING] [Treasury] " + colorize(message)
-        );
+        if (BukkitVendor.isPaper()) {
+            plugin
+                    .getServer()
+                    .getConsoleSender()
+                    .sendMessage(colorize("&e[WARNING] [Treasury] " + message));
+        } else {
+            plugin.getLogger().warning(colorize(message));
+        }
     }
 
     @Override
     public void error(String message) {
-        plugin.getServer().getConsoleSender().sendMessage("&4[ERROR] [Treasury]" + colorize(message));
+        if (BukkitVendor.isPaper()) {
+            plugin
+                    .getServer()
+                    .getConsoleSender()
+                    .sendMessage(colorize("&4[ERROR] [Treasury]" + message));
+        } else {
+            plugin.getLogger().severe(colorize(message));
+        }
     }
 
     @Override
     public void error(String message, Throwable t) {
-        plugin.getLogger().log(Level.SEVERE, colorize(message), t);
+        // in that case we're going to strip the colors (if any)
+        // currently this method isn't used for sending messages with colors but just to be on
+        // the right side let's strip if anything is there.
+        if (message.indexOf('&') != -1 || message.indexOf(ChatColor.COLOR_CHAR) != -1) {
+            message = ChatColor.stripColor(message.replace('&', ChatColor.COLOR_CHAR));
+        }
```

Comment:
This code will cause `&` characters to be replaced with `§` in messages, regardless if they are used for chat coloring.

Rest of the commit looks fantastic!

Lastly, I'm unfamiliar with the reason why we are stripping colors here, please inform me. :)

Reply:
It may be worth removing any consideration of the ampersand character in this method, since we only parse either non-colored or colorized inputs to this method anyways.

Reply:
> This code will cause `&` characters to be replaced with `§` in messages, regardless if they are used for chat coloring.

I know. I wanted to avoid parsing colors and then stripping them.

> Lastly, I'm unfamiliar with the reason why we are stripping colors here, please inform me. :)

this is a logging method that also attaches a exception so why shall we have colors here? 

> It may be worth removing any consideration of the ampersand character in this method, since we only parse either non-colored or colorized inputs to this method anyways.

didn't understand what you were trying to say

Reply:
I'll slightly deviate from my original thought and say that after ensuring that no existing usages of this method use colored messages, then there is no need to have the color stripping system.

Colors helps highlight important things to the server owner (and anyone helping them out with issues). I think all log messages should allow color. Feel free to contest me on this if you still prefer otherwise.

Reply:
I agree that all messages should allow color but go find out a method of logging exceptions with a colored message.

Reply:
We can definitely suspend the idea for later – if we ever find the need to use colored messages for the exception error logging method, we can just add in the feature then.